test/tests/TileElementsView.cpp

```diff
@@ -116,9 +116,9 @@ template<typename T> bool CompareLists(const CoordsXY& pos)
 
 template<typename T> void CheckMapTiles()
 {
-    for (int x = 0; x < MAXIMUM_MAP_SIZE_TECHNICAL; ++x)
+    for (int y = 0; y < MAXIMUM_MAP_SIZE_TECHNICAL; ++y)
     {
-        for (int y = 0; y < MAXIMUM_MAP_SIZE_TECHNICAL; ++y)
+        for (int x = 0; x < MAXIMUM_MAP_SIZE_TECHNICAL; ++x)
         {
             auto pos = TileCoordsXY(x, y).ToCoordsXY();
```

Comment:
Did you try doing a single loop and calculating the xy coordinates from the loop index?

Reply:
The issue is with map_get_element_at, using ```y + (max * x)```, so to get linear access you have to go by y first.